src/components/header/Header.tsx

```diff
@@ -0,0 +1,82 @@
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import { appContext } from '../Context/Context';
+import { Error } from '../../types/TypeOfErrors';
+
+export const TodoHeader: React.FC = () => {
+  const {
+    todos,
+    setErrors,
+    addTodoTitle,
+    USER_ID,
+    setIsLoading,
+    isLoading,
+    toggleAllToCompletedTodos,
+  } = useContext(appContext);
+
+  const inputRef = useRef<null | HTMLInputElement>(null);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isLoading]);
+
+  const handleOnSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event?.preventDefault();
+
+    if (!query.trim()) {
+      setErrors(Error.CorrectTitle);
+
+      return;
+    }
+
+    addTodoTitle({
+      userId: USER_ID,
+      title: query.trim(),
+      completed: false,
+    }).then(() => setQuery(''))
+      .finally(() => {
+        setIsLoading(false);
+      });
```

Comment:
You set is loading in `addTodoTitle` function and remove it here, keep it in one place, if you don't need this state in another components keep it here